hack/jenkins/e2e.sh

```diff
@@ -231,3 +233,10 @@ if [[ "${E2E_DOWN,,}" == "true" ]]; then
     sleep 30
     go run ./hack/e2e.go ${E2E_OPT} -v --down
 fi
+
+# TODO(zml): We have a bunch of legacy Jenkins configs that are
+# expecting junit.xml to be in ${WORKSPACE} root and it's Friday
+# afternoon, so just put the junit report where it's expected.
+if [ -e ${ARTIFACTS}/junit.xml ]; then
+  cp ${ARTIFACTS}/junit.xml ${WORKSPACE}
```

Comment:
make this a symlink, and move it before cleanup so that it runs even if teardown fails?